dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -1469,8 +1475,6 @@ private ExecutionStatus getProcessInstanceState(ProcessInstance instance) {
                 || CollectionUtils.isNotEmpty(failList)
                 || !isComplementEnd()) {
                 return ExecutionStatus.STOP;
-            } else {
-                return ExecutionStatus.SUCCESS;
```

Comment:
Why to remove this else branch? The state from `READY_STOP` to `SUCCESS` is possible.

Reply:
I can't think of any scene where ```ready_stop``` needs to be changed to ```success```. Can you enlight me a bit? @caishunfeng

Reply:
I think when no kill or failure task, the processInstance state can from `ready_stop` to `success`.
Otherwise, if no match `STOP` state, it will always keep `READY_STOP`. What do you think?

Reply:
I think of a scenario. When a downstream user uses a dependent node, if the result of kill command is success, the dependent node will judge that the execution is successful. In fact, the execution fails, resulting in an error in the detection result of the dependent node.